### PR TITLE
Update to support automatically synchronizing access token between libraries

### PR DESCRIPTION
**Pull request by danhdevelop**
## What kind of change does this PR introduce?

Updated to support automatically synchronizing access token between gotrue-py, postgrest-py and storage-py

## What is the current behavior?

Developers need to manually get an access token from gotrue-py and pass it to postgrest-py, storage-py... to authorize the connection to the backend.

## What is the new behavior?

All libraries share an http client class which contains a static variable to share authentication information.

## Additional context

Shared access token allow developers to reduce the steps to call API
When the access token is expired, the new token after being refreshed will be automatically supplied to all other libraries.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ httpx = "^0.23.0"
 storage3 = "^0.5.2"
 supafunc = "^0.2.2"
 python-semantic-release = "7.34.6"
+faker = "^18.10.1"
```
Comment (anand2312) [inline pyproject.toml:26]: I think faker ought to be a dev dependency

**Review by anand2312 — COMMENTED** [review]
I don't see code for the issue mentioned in the title - just a comment in the meantime

File: pyproject.toml (hunk 1 above)
Comment (danhdevelop) [reply pyproject.toml:26]: yes, you are right @anand2312 the faker should be at dev dependency.

**Comment by silentworks** [comment]
Thank you for your attempt at fixing this issue. We've merge another PR which solves this issue. https://github.com/supabase-community/supabase-py/pull/560

**Pull request closed without merging**